feat: Amplify JS v6 Developer Preview Documentation (#5914)
aws-amplify/docs: src/fragments/lib-v1/storage/js/download.mdx

The following options are available for retrieving previously uploaded data:

## Generate a download URL

Get a presigned URL of a stored file and expiration of URL. You can specify [access `level`](https://docs.amplify.aws/lib-v1/storage/configureaccess/q/platform/js) of file

```typescript
import { getUrl } from 'aws-amplify/storage';

const getUrlResult = await getUrl({
    key: filename,

>     options: { accessLevel?: 'guest' , // can be 'private' | 'protected' | 'guest' but defaults to `guest` 
>     targetIdentityId?: 'XXXXXXX', // id of another user, if `level: protected`
>     validateObjectExistence?: false,  // defaults to false
>     expiresIn?: 20 // validity of the URL, in seconds. defaults to 900 (15 minutes) and maxes at 3600 (1 hour)
>     useAccelerateEndpoint?: true; // Whether to use accelerate endpoint.
>     },
> });

Reviewer: ```suggestion
    options: {
        accessLevel?: 'guest' , // can be 'private', 'protected', or 'guest' but defaults to `guest`
        targetIdentityId?: 'XXXXXXX', // id of another user, if `accessLevel` is `guest`
        validateObjectExistence?: false,  // defaults to false
        expiresIn?: 20 // validity of the URL, in seconds. defaults to 900 (15 minutes) and maxes at 3600 (1 hour)
        useAccelerateEndpoint?: true; // Whether to use accelerate endpoint.
    },
});
```

```javascript
import { getUrl } from 'aws-amplify/storage';

// To check for existence of a file
await getUrl({
	key: filename,
	options: {
		validateObjectExistence: true, // defaults to false
	},
});

```

### getUrl expiry

You can use `expiresIn` option to limit the availability of your URLs. This configuration returns the pre-signed URL that expires in 60 seconds:

```javascript
import { getUrl } from 'aws-amplify/storage';

await getUrl({ key: 'filename.txt', options: { expiresIn: 60 } });
```

## downloadData

Download a file to in-memory buffer.

```javascript
import { downloadData } from 'aws-amplify/storage';

// Download a file from s3 bucket
const { body, eTag } = await downloadData({
	key,
	data: file,
	options: {
		accessLevel: 'guest', // can be 'private' | 'protected' | 'guest'
		targetIdentityId: 'xxxxxxx', // id of another user, if `level: protected`
		onProgress, // Optional progress callback.
	},
}).result;
 ```
 
### Get the text value of downloaded File

> we can get  the value of file in any of the three 'blob' | 'json' | 'text' formats. Replaces the respective method to get respective value.

Reviewer: ```suggestion
we can get  the value of file in any of the three formats: 'blob', 'json', 'text'. Replaces the respective method to get respective value.
```

```javascript
import { downloadData } from 'aws-amplify/storage';

try {
	const downloadResult: { body: Pick<Body, 'blob' | 'json' | 'text'> } 
	= await downloadData({ key: filename }).result;
	if (downloadResult?.body?.text)
		downloadResult?.body
			?.text()
			.then((result) => console.log(result));
} catch (error) {
	console.log('Error : ', error)
}
```

> To track the progress of your download, you can use `progressCallback`:

Reviewer: ```suggestion
To track the progress of your download, you can use `onProgress`:
```
